apps/settings/test/unit/panels/call_barring/call_barring_test.js

```diff
@@ -0,0 +1,393 @@
+/* global MockMobileconnection */
+'use strict';
+
+require('/shared/test/unit/mocks/mock_navigator_moz_mobile_connections.js');
+
+suite('Call Barring settings', function() {
+  var realMozMobileConnection;
+
+  var _mobileConnection,
+      _serviceClass;
+
+  var callBarring;
+
+  suiteSetup(function(done) {
+
+    realMozMobileConnection = navigator.mozMobileConnection;
+    navigator.mozMobileConnection = MockMobileconnection;
+
+    var modules = [
+      'panels/call_barring/call_barring'
+    ];
+
+    var maps = {
+      'panels/call_barring/call_barring': {
+        'modules/settings_service': 'unit/mock_settings_service'
+      }
+    };
+
+    testRequire(modules, maps, function(CallBarring) {
+      callBarring = CallBarring;
+      done();
+    });
+
+  });
+
+  suiteTeardown(function() {
+    navigator.mozMobileConnection = realMozMobileConnection;
+  });
+
+  setup(function() {
+    _mobileConnection = navigator.mozMobileConnection();
+    _serviceClass = _mobileConnection.ICC_SERVICE_CLASS_VOICE;
+  });
+
+  suite('> GET data', function() {
+
+    var _cbServiceMapper = {
+      'li-cb-baoc': 0,
+      'li-cb-boic': 1,
+      'li-cb-boic-exhc': 2,
+      'li-cb-baic': 3,
+      'li-cb-baic-r': 4
+    };
+
+    var serviceOn = {
+      set onsuccess(cb) {
+        this.result = {
+          'enabled': true
+        };
+        cb.call(this);
+      }
+    };
+    var serviceOff = {
+      set onsuccess(cb) {
+        this.result = {
+          'enabled': false
+        };
+        cb.call(this);
+      }
+    };
+
+    setup(function() {
+      _mobileConnection.getCallBarringOption = sinon.stub();
```

Comment:
We could use `this.sinon.stub(_mobileConnection, 'getCallBarringOption');`, then we don't have to restore it in the teardown function.